src/templates/Practice4.tsx

```diff
@@ -0,0 +1,132 @@
+    postalCode: string()
+      .required('必須項目です')
+      .matches(/^\d{7}$/, '7桁の数字を入力してください'),
+    prefecture: string().required('必須項目です'),
+    city: string()
+      .required('必須項目です'),
+  }).required();
+
+type FormData = InferType<typeof validationSchema>;
+
+const Practice4: React.FC = () => {
+  const [canSubmit, setCanSubmit] = useState<Boolean>(true);
+  const [canSearch, setCanSearch] = useState<Boolean>(true);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }, getValues,setValue,trigger,
+  } = useForm<FormData>({
+    resolver: yupResolver(validationSchema),
+    defaultValues: {
+      postalCode: '',
+      prefecture: '',
+      city: '',
+    },
+    mode: 'onChange',
+  });
+
+  // Form submission handler
+  const onSubmit = handleSubmit(data => {
+    console.log(data);
+  });    
+
+  const onSearch = async() => 
+    {
+      setValue("prefecture",'');
+      setValue("city",'');
+      const pCodeVal = getValues('postalCode');
+      if(!pCodeVal.trim())
+        setCanSearch(false);
+
+      else if(pCodeVal.length===7)
+      { 
+        try
+        {
+          const response = await fetch("https://zipcloud.ibsnet.co.jp/api/search?zipcode="+pCodeVal);
+          const fields = await response.json();
+          // console.log(fields);
+          setValue("postalCode",pCodeVal);
+          setValue("prefecture",fields.results[0].address1);
+          setValue("city",fields.results[0].address2);
+          setCanSubmit(true);
+        }
+
+        catch (error) {
+          setValue('city', '');
+          setValue('prefecture', '');
+          setCanSubmit(false);
+        } finally {
+          trigger();
+        }
+        setCanSearch(true);
+      }
+    };
+  return (
+    <div className="m-auto flex flex-col gap-4 ...">
+      <h1 className="text-4xl text-center mt-2">addressSearch</h1>
+      <form onSubmit={onSubmit} className="m-auto flex flex-col gap-4" noValidate>
+        <div className="relative flex flex-row gap-4">
+          <label className="font-bold" htmlFor="postalCode">郵便番号</label>
+          <div className="relative flex flex-row gap-2">
+          <div className="relative">
+          <input
+            id="postalCode" type="text" className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-60 p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:focus:ring-blue-500"
+            {...register('postalCode')}
+          />
+          {errors.postalCode && (
+            <p className="text-xs italic text-red-500">{errors.postalCode.message}</p>
+          )}
+          </div>
+          <div className="relative">
+          <Button title="search" onClick={onSearch} className=' bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded'>
+          Search
+          </Button>
+          {canSearch? (null):(<p className="text-xs italic text-red-500">必須項目です</p>)}
```

Comment:
![image](https://github.com/user-attachments/assets/0187ad15-6912-46ef-8b35-ef886515c70b)

I think having the left error message under the input is enough. We don't need to display the same message twice.

Reply:
The way you told yesterday worked ! Have made the changes